src/WebJobs.Extensions.DocumentDB/Trigger/CosmosDBTriggerAttributeBindingProvider.cs

```diff
@@ -91,27 +92,55 @@ public Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext contex
 
         private string ResolveAttributeConnectionString(CosmosDBTriggerAttribute attribute)
         {
-            if (string.IsNullOrEmpty(_monitorConnectionString) &&
-                string.IsNullOrEmpty(attribute.ConnectionStringSetting))
+            string connectionString = this._monitorConnectionString;
+            if (!TryReadFromSettings(attribute.ConnectionStringSetting, connectionString, out connectionString))
+            {
+                throw new InvalidOperationException(
+                    string.Format(CultureInfo.CurrentCulture,
+                    "The CosmosDBTriggerAttribute.ConnectionStringSetting '{0}' property specified does not exist in the Application Settings.",
+                    attribute.ConnectionStringSetting));
+            }
+
+            if (string.IsNullOrEmpty(connectionString))
```

Comment:
Will you ever get a null or empty here? Doesn't `TryReadFromSettings()` check that above? Also, this error message is more complete -- if you remove this `if` block completely, can you re-use the message here in the check above?